File: python/repetição/maior_soma_media.py
print("---------------------MAIOR, SOMA, MÉDIA----------------------\n")

su = 0
average = 0
max_value = 0
value = 0

value = int(input("\nDigite o 1º valor: "))

while value < 0:
    print("ERROR!!! O número pode apenas ser positivo!!!")
    value = int(input("\nDigite o 1º valor: "))

max_value = value
su+=value

for i in range(2, 11):
    
    value = int(input(f"\nDigite o {i} valor: "))
    
    while num_values <= 0:
       print("ERROR!!! O número pode apenas ser positivo!!!")
       num_values = int(input("\nQuantos valores você quer digitar? "))

    while value < 0:
        print("ERROR!!! O número pode apenas ser positivo!!!")
        value = int(input("\nDigite o {i} valor: "))
        
    if value > max_value:
        max_value = value
    
    su+=value

average = su / 10;

print("------------------------------------------------------")
print(f"O maior valor digitado foi {max_value}")
print(f"A soma dos valores é igual a {su}")
print(f"A média aritmética dos valores é igual a {average}")
  

File: java/repeticao/MaiorSomaMedia.java
package repeticao;

import java.util.Scanner;

import libraries.TerminalFormating;
import libraries.Validation;

public class MaiorSomaMedia {
    public static void main(String[] args) {

        double sum = 0;
        double average = 0;
        double max_value = 0;
        double value = 0;
        
        Scanner input = new Scanner(System.in);
        TerminalFormating terminal = new TerminalFormating();
        Validation validation = new Validation();

        //inicio do programa

        terminal.writeProgramTitle("MAIOR, SOMA E MÉDIA");

        System.out.print("Digite o 1º valor: " );
        value = input.nextDouble();

        while(validation.checkIfNumberIsnegative(value)) {
            terminal.errorMessage("O valor não pode ser negativo!");
            System.out.print("Digite o 1º valor: " );
            value = input.nextDouble();
        }

        max_value = value;
		sum+=value;

        for (int i = 2; i <= 10; i++) {
            System.out.print("Digite o " + i + "º valor: " );
            value = input.nextDouble();

            while(validation.checkIfNumberIsnegative(value)) {
                terminal.errorMessage("O valor não pode ser negativo!");
                System.out.print("Digite o " + i + "º valor: " );
                value = input.nextDouble();
            }

            if (value > max_value) {
                max_value = value;
            }

            sum+= value;
        }

        average = sum / 10;

        System.out.println("O maior valor digitado foi " + max_value);
        System.out.println("A soma dos valores é igual a " + sum);
        System.out.println("A média aritmética dos valores é igual a " + average);
    }
}
